src/block_cache/block_cache.c

```diff
@@ -528,7 +529,14 @@ __wt_blkcache_remove(WT_SESSION_IMPL *session, const uint8_t *addr, size_t addr_
             WT_STAT_CONN_INCR(session, block_cache_blocks_removed);
             (void)__wt_atomic_sub64(&blkcache->bytes_used, blkcache_item->data_size);
             blkcache->removals++;
-            WT_ASSERT(session, blkcache_item->ref_count == 0);
+            /*
+             * The block might be in use by another thread, wait for it to be released before
+             * freeing it.
+             */
+            while (blkcache_item->ref_count != 0) {
+                __wt_spin_backoff(&yield_count, &sleep_usecs);
+                WT_STAT_CONN_INCRV(session, block_cache_blocks_removed_blocked, sleep_usecs);
```

Comment:
I thought @mm-ng 's suggestion was to increment the statistic once at the end rather than every loop iteration. I think that would be better. So have something like `total_usecs` that you increment by `sleep_usecs` here, and then once the loop completes we can INCRV by `total_usecs`.

Reply:
@mm-ng, can you confirm? I understood that we should do all the other calls, then check for `ref_count` at last before calling `__blkcache_free`, is that correct, or is it what @sueloverso is saying?
Nonetheless, I am happy to do this too @sueloverso.

Reply:
@etienneptl Initially I was thinking of just moving the statistics increment to after we assert, but I actually think Sue's approach could work better and I'm happy to go with that as well.